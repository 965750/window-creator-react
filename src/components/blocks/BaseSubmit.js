import React from 'react'
import { injectIntl } from 'react-intl'
import PropTypes from 'prop-types'
import styles from './BaseSubmit.module.scss'

const BaseSubmit = ({ disabled, value, intl }) => (
    <input
        disabled={disabled}
        type="submit"
        className={`${styles.submit} cursor-pointer block bg-first rounded text-white mx-auto`}
        value={intl.formatMessage({
            id: value,
        })}
    />
)

BaseSubmit.propTypes = {
    disabled: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
    value: PropTypes.string,
}

export default injectIntl(BaseSubmit)
