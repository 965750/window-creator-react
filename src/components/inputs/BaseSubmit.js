import { injectIntl } from 'react-intl'

import PropTypes from 'prop-types'
import React from 'react'
import styles from './BaseSubmit.module.scss'

const BaseSubmit = ({ disabled, value, intl }) => (
  <input
    disabled={disabled}
    type="submit"
    className={`${styles.submit} cursor-pointer block bg-mountainMeadow rounded text-white mx-auto`}
    value={intl.formatMessage({
      id: value,
    })}
  />
)

BaseSubmit.propTypes = {
  disabled: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  value: PropTypes.string,
  intl: PropTypes.shape({
    formatMessage: PropTypes.func,
  }),
}

export default injectIntl(BaseSubmit)
