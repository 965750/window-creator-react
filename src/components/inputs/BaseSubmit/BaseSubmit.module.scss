@import '../../../variables';

.submit {
  height: 50px;
  width: 400px;
  outline: none;

  @media screen and (max-width: 640px) {
    width: 300px;
  }

  &--sm {
    width: 100px;
    min-width: 100px;
  }

  &:disabled {
    background: $gullGray;
  }
}
