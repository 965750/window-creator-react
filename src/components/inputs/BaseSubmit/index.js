import { injectIntl } from 'react-intl'

import PropTypes from 'prop-types'
import React from 'react'
import styles from './BaseSubmit.module.scss'

const BaseSubmit = ({
  disabled, value, intl, sm, classes,
}) => (
  <input
    data-testid="submitWrapper"
    disabled={disabled}
    type="submit"
    className={`${styles.submit} ${sm ? styles['submit--sm'] : ''} ${classes || ''} cursor-pointer block bg-mountainMeadow rounded text-white`}
    value={intl.formatMessage({
      id: value,
    })}
  />
)

BaseSubmit.propTypes = {
  disabled: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  value: PropTypes.string,
  intl: PropTypes.shape({
    formatMessage: PropTypes.func,
  }),
  sm: PropTypes.bool,
  classes: PropTypes.string,
}

export default injectIntl(BaseSubmit)
