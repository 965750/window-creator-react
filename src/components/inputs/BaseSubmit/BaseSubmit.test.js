import '@testing-library/jest-dom/extend-expect'

import { applyMiddleware, compose, createStore } from 'redux';
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { render, cleanup } from '@testing-library/react'

import BaseSubmit from './index'
import messages from '../../../messages-i18n'
import thunk from 'redux-thunk';
import React from 'react'

const initialStore = {
  theme: {
    lang: 'pl',
  },
}

const reducer = (state = initialStore, action) => {
  switch (action.type) {
    default:
      return state
  }
}

const store = createStore(
  reducer,
  compose(
    applyMiddleware(thunk),
  ),
)

const renderWithRedux = (component) => ({
  ...render(<Provider store={store}>{component}</Provider>),
})

describe('BaseSubmit', () => {
  let wrapper
  const initialProps = {
    disabled: true,
    value: 'Login'
  }

  const setUp = (props = initialProps) => {
    wrapper = renderWithRedux(
      <IntlProvider
        locale="en"
        messages={messages[store.getState().theme.lang]}
      >
        <BaseSubmit {...props} />
      </IntlProvider>,
    )
  }

  afterEach(cleanup)

  it('should match snapshot', () => {
    setUp()

    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('should pass classes from props to wrapper', () => {
    const mockClasses = 'mockClasses'

    setUp({
      ...initialProps,
      classes: mockClasses,
    })

    expect(wrapper.queryByTestId('submitWrapper')).toHaveClass(mockClasses)
  })

  it('should get class when sm prop was added', () => {
    setUp({
      ...initialProps,
      sm: true,
    })

    expect(wrapper.queryByTestId('submitWrapper')).toHaveClass('submit--sm')
  })
})
