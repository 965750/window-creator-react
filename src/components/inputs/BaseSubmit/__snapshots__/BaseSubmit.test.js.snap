// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseSubmit should match snapshot 1`] = `
<DocumentFragment>
  <input
    class="submit   cursor-pointer block bg-mountainMeadow rounded text-white"
    data-testid="submitWrapper"
    disabled=""
    type="submit"
    value="Zaloguj"
  />
</DocumentFragment>
`;
